qiita_db/artifact.py

```diff
@@ -458,11 +458,11 @@ def _associate_with_analysis(instance, analysis_id):
                 visibilities = {a.visibility for a in instance.parents}
                 # set based on the "lowest" visibility
                 if 'sandbox' in visibilities:
-                    instance.visibility = 'sandbox'
+                    instance._set_visibility('sandbox')
```

Comment:
Some background: when an artifact is created, it will "copy" the parents visibility; however, right now the visibility "regular" setter (visibility) has a validation step that prevents this from happening. Thus, the "solution" is that for creation we use the new setter (_set_visibility) without validation and when we do whatever else we use the regular one, which has the validation ... AKA special case for artifact creation